### PR TITLE
poac: 0.5.1 -> 0.10.1

### PR DESCRIPTION
**Pull request by eihqnh**
## Description of changes
1. waiting for https://github.com/NixOS/nixpkgs/pull/331649 
2. [update poac ](poac: 0.5.1 -> 0.10.1)
3. add me as the maintainer
4. move to pkgs/by-name

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by symphorien** [comment]
Does not build for me on x86-64-linux
```
these 2 derivations will be built:
  /nix/store/j233j3wjp48bmaa3drlvcgin8yqqwsrm-source.drv
  /nix/store/rqkxj3fvxycwfihldxacqygrsnq5jr7k-poac-0.10.1.drv
building '/nix/store/j233j3wjp48bmaa3drlvcgin8yqqwsrm-source.drv'...
exporting https://github.com/poac-dev/poac.git (rev refs/tags/v0.10.1) into /nix/store/lv14b5dvzrbydr0xxxk2s4nqq6kn7f32-source
Initialized empty Git repository in /nix/store/lv14b5dvzrbydr0xxxk2s4nqq6kn7f32-source/.git/
fatal: couldn't find remote ref refs/tags/v0.10.1
remote: Enumerating objects: 156, done.
remote: Counting objects: 100% (156/156), done.
remote: Compressing objects: 100% (143/143), done.
remote: Total 156 (delta 8), reused 80 (delta 6), pack-reused 0 (from 0)
Receiving objects: 100% (156/156), 126.76 KiB | 3.96 MiB/s, done.
Resolving deltas: 100% (8/8), done.
From https://github.com/poac-dev/poac
 * branch            HEAD       -> FETCH_HEAD
fatal: Not a valid object name 
Unrecognized git object type: 
Unable to checkout refs/tags/v0.10.1 from https://github.com/poac-dev/poac.git.
```

**Comment by qwqawawow** [comment]
> Does not build for me on x86-64-linux
> 
> ```
> these 2 derivations will be built:
>   /nix/store/j233j3wjp48bmaa3drlvcgin8yqqwsrm-source.drv
>   /nix/store/rqkxj3fvxycwfihldxacqygrsnq5jr7k-poac-0.10.1.drv
> building '/nix/store/j233j3wjp48bmaa3drlvcgin8yqqwsrm-source.drv'...
> exporting https://github.com/poac-dev/poac.git (rev refs/tags/v0.10.1) into /nix/store/lv14b5dvzrbydr0xxxk2s4nqq6kn7f32-source
> Initialized empty Git repository in /nix/store/lv14b5dvzrbydr0xxxk2s4nqq6kn7f32-source/.git/
> fatal: couldn't find remote ref refs/tags/v0.10.1
> remote: Enumerating objects: 156, done.
> remote: Counting objects: 100% (156/156), done.
> remote: Compressing objects: 100% (143/143), done.
> remote: Total 156 (delta 8), reused 80 (delta 6), pack-reused 0 (from 0)
> Receiving objects: 100% (156/156), 126.76 KiB | 3.96 MiB/s, done.
> Resolving deltas: 100% (8/8), done.
> From https://github.com/poac-dev/poac
>  * branch            HEAD       -> FETCH_HEAD
> fatal: Not a valid object name 
> Unrecognized git object type: 
> Unable to checkout refs/tags/v0.10.1 from https://github.com/poac-dev/poac.git.
> ```

Thank you very much. I had fixed that.

File: pkgs/by-name/po/poac/package.nix [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    substituteInPlace Makefile \
+       --replace-fail "git clone" "\#git clone"
+    substituteInPlace Makefile \
+       --replace-fail "git -C" "\#git -c"
+  '';
+  buildPhase = ''
+    mkdir -p build-out/DEPS/
+    cp -rf ${toml11} build-out/DEPS/toml11
+    make RELEASE=1
+  '';
```
Comment (symphorien) [inline pkgs/by-name/po/poac/package.nix:57]:
```suggestion
  preBuild = ''
    mkdir -p build-out/DEPS/
    cp -rf ${toml11} build-out/DEPS/toml11
  '';
  makeFlags = [ "RELEASE=1" ];
```

overriding the whole phase disables preBuild and postBuild. you also miss automation like enableParallelBuilding

File: pkgs/by-name/po/poac/package.nix [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  ];
+  buildInputs = [
+    libgit2
+    fmt
+    tbb_2021_11
+    nlohmann_json
+    curl
+    git
+  ];
+  configurePhase = ''
```
Comment (symphorien) [inline pkgs/by-name/po/poac/package.nix:46]: see below, please don't override full phases.

File: pkgs/by-name/po/poac/package.nix [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+       --replace-fail "git -C" "\#git -c"
+  '';
+  buildPhase = ''
+    mkdir -p build-out/DEPS/
+    cp -rf ${toml11} build-out/DEPS/toml11
+    make RELEASE=1
+  '';
+  installPhase = ''
+    make install PREFIX=$out
+  '';
```
Comment (symphorien) [inline pkgs/by-name/po/poac/package.nix:60]:
```suggestion
  installFlags = [ "PREFIX=${placeholder "out"}" ];
```

File: pkgs/by-name/po/poac/package.nix [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  nativeBuildInputs = [
+    pkg-config
+  ];
+  buildInputs = [
+    libgit2
+    fmt
+    tbb_2021_11
+    nlohmann_json
+    curl
+    git
```
Comment (symphorien) [inline pkgs/by-name/po/poac/package.nix:44]: the output does not reference git. Is it really a buildInput ? if it is only used as an executable during build, then add it to nativeBuildInputs. setting `strictDeps=true;` (which I recommend) avoids mixing buildInputs and nativeBuildInputs.

**Review by symphorien — COMMENTED** [review]
thank you for the fix. I left a few further comments.

**Comment by qwqawawow** [comment]
Thank you for your review and guidance.

**Comment by symphorien** [comment]
Thank you for the changes.

Please change your PR to only contain one commit whose message adheres to the guidelines: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions

I also have a suggestion, if it is doable:
you use the leaveDotGit option of fetchFromGithub, which should really be avoided, because despite all the work it does it does there is no guarantee that the downloaded source will be deterministic. You use this option so that the executable can embed its version. The makefile presumably calls git to generate a C file with the version as a constant. If possible you should replace that by creating the file yourself in preConfigure with the nix variable `${version}` like here:
https://github.com/NixOS/nixpkgs/blob/nixos-24.05/pkgs/servers/osmocom/libasn1c/default.nix#L20-L22

Then you can remove the leaveDotGit option (and change the hash of the source).

**Comment by qwqawawow** [comment]
> Thank you for the changes.
> 
> Please change your PR to only contain one commit whose message adheres to the guidelines: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions
> 
> I also have a suggestion, if it is doable: you use the leaveDotGit option of fetchFromGithub, which should really be avoided, because despite all the work it does it does there is no guarantee that the downloaded source will be deterministic. You use this option so that the executable can embed its version. The makefile presumably calls git to generate a C file with the version as a constant. If possible you should replace that by creating the file yourself in preConfigure with the nix variable `${version}` like here: https://github.com/NixOS/nixpkgs/blob/nixos-24.05/pkgs/servers/osmocom/libasn1c/default.nix#L20-L22
> 
> Then you can remove the leaveDotGit option (and change the hash of the source).

done.Thank you very much. 
I squashed it. 
It is a bit difficult to get commit date for me.
Any suggestions?

**Comment by qwqawawow** [comment]
> I would leave it as 1970-01-01. Alternatively you can hard code it in the nix file, with the expectation that you change it on every version bump. Random example https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/di/digital/package.nix#L9

Thank you very much.I think there is no need to track the date.

**Review by symphorien — APPROVED** [review]

**Comment by symphorien** [comment]
Hum I notice that poac fails to build on darwin. Can you either fix this or mark it as broken on this platform ? (Like this: https://github.com/NixOS/nixpkgs/blob/master/pkgs/misc/jackaudio/jack1.nix#L38 )

**Comment by qwqawawow** [comment]
> Hum I notice that poac fails to build on darwin. Can you either fix this or mark it as broken on this platform ? (Like this: https://github.com/NixOS/nixpkgs/blob/master/pkgs/misc/jackaudio/jack1.nix#L38 )

done
Looks like broken on x86-darwin
https://github.com/qwqawawow/nixpkgs-test/actions/runs/12106089993/job/33751464640
[master rev 13c672](https://github.com/NixOS/nixpkgs/commit/13c6725c19125c43de87e0579214d2cbef4426b4)
Thank you for your guidance. I feel ashamed for not carefully reading the manual and am deeply grateful.

**Pull request merged**